discovery_engine_core/web-api/src/bin/ingestion.rs

```diff
@@ -211,7 +211,7 @@ async fn handle_add_data(
                 ElasticDocumentData {
                     snippet: document.snippet,
                     properties: document.properties,
-                    embedding,
+                    embedding: embedding.to_vec(),
```

Comment:
:wrench: `to_vec()` always clones the data, which can become quite expensive since we ha a lot of arrays. we can add a utility funciton in the `xayn_discovery_engine_bert::pooler` module in the `impl Embedding<Ix1>` block:
```rust
    pub fn into_vec(self) -> Vec<f32> {
        if self.is_standard_layout() {
            self.0.into_raw_vec()
        } else {
            self.to_vec()
        }
    }
```
this way it's only cloned if the data is not in standard layout, which usually should be the cold path.